lib/puppet.rb

```diff
@@ -19,6 +19,20 @@
 require 'puppet/external/pson/version'
 require 'puppet/external/pson/pure'
 
+## It would be really nice to use Puppet.features here, but we have to
+## do gettext setup before all of that machinery is loaded. Instead,
+## we will use a simple boolean to handle gettext availability in this
+## file, and drop in a "stub" translation function here.
+begin
```

Comment:
Is there a longer term plan for ensuring gettext functions are available so that we can remove this?